docs/content/TreeView.mdx

```diff
@@ -98,6 +133,54 @@ function ControlledTreeView() {
 render(<ControlledTreeView />)
 ```
 
+### With stateful visuals
+
+To render stateful visuals, pass a render function to `TreeView.LeadingVisual` and `TreeView.TrailingVisual`. The function will be called with the `expanded` state of the item.
+
+```jsx live drafts
+<Box sx={{maxWidth: 400}}>
+  <nav aria-label="File navigation">
+    <TreeView aria-label="File navigation">
+      <TreeView.Item>
+        <TreeView.LeadingVisual>
+          {({isExpanded}) => (isExpanded ? <FileDirectoryOpenFillIcon /> : <FileDirectoryFillIcon />)}
+        </TreeView.LeadingVisual>
+        src
+        <TreeView.SubTree>
+          <TreeView.LinkItem href="#">Avatar.tsx</TreeView.LinkItem>
+          <TreeView.LinkItem href="#" current>
+            Button.tsx
+          </TreeView.LinkItem>
+        </TreeView.SubTree>
+      </TreeView.Item>
+    </TreeView>
+  </nav>
+</Box>
+```
+
+Since stateful directory icons are a common usecase for TreeView, we provide a `TreeView.DirectoryIcon` component for convenience. The above example can be rewritten as:
```

Comment:
```suggestion
Since stateful directory icons are a common use case for TreeView, we provide a `TreeView.DirectoryIcon` component for convenience. The previous example can be rewritten as:
```